### ISSUE TITLE
Was unable to delete characters from my message

### ISSUE THREAD

**Issue by turt2live** [vector-im/riot-web#6420]
### Description

![image](https://user-images.githubusercontent.com/1190097/38123159-e4827a2e-3396-11e8-98ff-b3ba98494fea.png)

Tried:
* Backspace 
* Delete key
* Highlighting, then using delete or backspace
* Mashing the backspace key harder
* Opening the console to find an error

I originally wanted to remove the "r" at the end of requests, then realized I couldn't, mashed the keyboard to add some characters, and couldn't remove them. Had to refresh the page. Switching rooms was not good enough.

```
Uncaught TypeError: Cannot read property 'getEntityAt' of undefined
    at Object.removeRange (DraftModifier.js:100)
    at removeTextWithStrategy (removeTextWithStrategy.js:39)
    at keyCommandPlainDelete (keyCommandPlainDelete.js:27)
    at onKeyCommand (editOnKeyDown.js:46)
    at editOnKeyDown (editOnKeyDown.js:141)
    at DraftEditor.react.js:146
    at HTMLUnknownElement.boundFunc (ReactErrorUtils.js:63)
    at Object.ReactErrorUtils.invokeGuardedCallback (ReactErrorUtils.js:69)
    at executeDispatch (EventPluginUtils.js:83)
    at Object.executeDispatchesInOrder (EventPluginUtils.js:106)
    at executeDispatchesAndRelease (EventPluginHub.js:41)
    at executeDispatchesAndReleaseTopLevel (EventPluginHub.js:52)
    at Array.forEach (<anonymous>)
    at forEachAccumulated (forEachAccumulated.js:22)
    at Object.processEventQueue (EventPluginHub.js:252)
    at runEventQueueInBatch (ReactEventEmitterMixin.js:15)
    at Object.handleTopLevel [as _handleTopLevel] (ReactEventEmitterMixin.js:25)
    at handleTopLevelImpl (ReactEventListener.js:70)
    at ReactDefaultBatchingStrategyTransaction.perform (Transaction.js:141)
    at Object.batchedUpdates (ReactDefaultBatchingStrategy.js:60)
    at Object.batchedUpdates (ReactUpdates.js:95)
    at dispatchEvent (ReactEventListener.js:145)
```

### Version information

- **Platform**: web (in-browser)
- **Browser**: Chrome 64
- **OS**: Windows 10
- **URL**: riot.im/develop

**Comment by WGH-** [comment]
I've seen similar (if not the same) bug many times on Firefox too. I don't have reproduction case. It has always occurred randomly, and _probably_ caused by selecting (e.g. Ctrl+A) and deleting text in some conditions.

In my case, once input text box enters the failure mode, the characters will be "actually" deleted, but they appear visually. In fact, if you hit Backspace enough times, "Send message" prompt in gray will appear below this "undeletable" text.

**Comment by lampholder** [comment]
This dosen't seem to be super common but _is_ super annoying.

**Comment by ara4n** [comment]
fixed by matrix-org/matrix-react-sdk#1890
